vm-c/vm.cpp

```diff
@@ -0,0 +1,248 @@
+#include <gmp.h>
+#include <iostream>
+#include "proto/msg.pb.h"
+#include "runtime/header.h"
+#include "runtime/alloc.h"
+#include "vm.h"
+#include "semantics.h"
+#include "world.h"
+
+using namespace io::iohk::ethereum::extvm;
+
+std::vector<mpz_ptr> k_to_z(list* l) {
+  std::vector<mpz_ptr> result;
+  for (size_t i = 0; i < hook_LIST_size_long(l); i++) {
+     zinj* elem = (zinj*) hook_LIST_get_long(l, i);
+     result.push_back(elem->data);
+  }
+  return result;
+}
+
+std::vector<account *> k_to_acct(map* m) {
+  list l = hook_MAP_values(m);
+  std::vector<account *> result;
+  for (size_t i = 0; i < hook_LIST_size_long(&l); i++) {
+     account* elem = (account*) hook_LIST_get_long(&l, i);
+     result.push_back(elem);
+  }
+  return result;
+}
+
+std::vector<log *> k_to_log(list* l) {
```

Comment:
I'd prefer plural names for these functions parsing a K list to a vector. Especially to avoid overloading just on signature between the single-entry and multiple-entry versions of `k_to_log`.